src/avram/nothing.cr

```diff
@@ -0,0 +1,5 @@
+# This class is used in various places
+# where the question of "Did I not pass in anything or did I pass in nil?"
+# needs to be answered.
+class Avram::Nothing
```

Comment:
Lets add a `:nodoc:` here so this doesn't get generated in the docs. This really should be a private class since it's not meant to be used externally, but making it `private` I think would break..